crates/core/tedge/src/system_services/managers/general_manager.rs

```diff
@@ -0,0 +1,399 @@
+use crate::system_services::{
+    CommandBuilder, InitConfig, SystemConfig, SystemService, SystemServiceError,
+    SystemServiceManager, SERVICE_CONFIG_FILE,
+};
+use std::fmt;
+use std::path::PathBuf;
+use std::process::ExitStatus;
+use tedge_users::{UserManager, ROOT_USER};
+
+#[derive(Debug)]
+pub struct GeneralServiceManager {
+    user_manager: UserManager,
+    init_config: InitConfig,
+    config_path: String,
+}
+
+impl GeneralServiceManager {
+    pub fn try_new(
+        user_manager: UserManager,
+        config_root: PathBuf,
+    ) -> Result<Self, SystemServiceError> {
+        let init_config = SystemConfig::try_new(config_root.clone())?.init;
+        let config_path = config_root
+            .join(SERVICE_CONFIG_FILE)
+            .to_str()
+            .unwrap()
+            .to_string(); // Path always exists
```

Comment:
The `unwrap()` is not related to the file existence but to whether or not the path is UTF8.

```suggestion
        let config_path = config_root
            .join(SERVICE_CONFIG_FILE)
            .to_str()
            .unwrap_or(SERVICE_CONFIG_FILE); 
```